### PR TITLE
Fix issue with shortcut recompression

### LINKED ISSUE #352: Trailing comments aren't being properly transferred
While working on #351 I noticed that `cell.comments` was wrong. 

The following cell:

``` 
100103  3926  5.00000E-02 (-102002          -36425  36426):   $ Lower Rodlet Sodium Bond
                          (-102002  169239  -36426  36427)    u=424 
c
c comment for next cell... 
c
... next cell
```

Had too many comments in `cell.comments`

``` python
In [13]: cell.comments
Out[13]:
[COMMENT:
 $ Lower Rodlet Sodium Bond,
 COMMENT:
 c,
 COMMENT:
 c comment for next cell... 
 COMMENT:
 c,
```

**Additional context**
Version: `0.3.0dev1`

Note to self: see if next cell actually got parsed

### LINKED ISSUE #489: Botched Integer Interpolation Expansion
**Describe the bug**

MontePy expands shortcuts for data entry:

image

The linear interpolation shortcut gets botched upon expansion, being converted to floating point and dropping necessary padding.

**To Reproduce**

Just load an MCNP input file with the shortcut and write it back out.

``` python
import montepy
montepy.read_input("testi.i").write_to_file("testo.i")
```

**MCNP input file snippet**

If applicable, please include a small section of the input file you were working on. If it includes any specific values please change or remove them. For example:

```
MCNP Test Model
C Cells
C Seven spheres within a sphere
1 1 -10 -1001  imp:n,p=2
2 1 -10 -1002  imp:n,p=2
3 1 -10 -1003  imp:n,p=2
4 1 -10 -1004  imp:n,p=2
5 1 -10 -1005  imp:n,p=2
6 1 -10 -1006  imp:n,p=2
7 1 -10 -1007  imp:n,p=2
C This cell breaks MontePy 0.3.3
8     0 -1010  1001  5i  1007  imp:n,p=1
C Graveyard
99    0 +1010     imp:n,p=0

C surfaces
1001 SO      0.1
1002 SX -1.1 0.1
1003 SX +1.1 0.1
1004 SY -1.1 0.1
1005 SY +1.1 0.1
1006 SZ -1.1 0.1
1007 SZ +1.1 0.1
1010 SO 3

C data
C materials
C UO2 5 atpt enriched
m1        92235.80c           5
          92238.80c          95
C execution
ksrc 0 0 0
kcode 100000 1.000 50 1050
mode n p

```
**Error Message (if any)**

```
MCNP Test Model
C Cells
C Seven spheres within a sphere
1 1 -10 -1001  imp:n,p=2
2 1 -10 -1002  imp:n,p=2
3 1 -10 -1003  imp:n,p=2
4 1 -10 -1004  imp:n,p=2
5 1 -10 -1005  imp:n,p=2
6 1 -10 -1006  imp:n,p=2
7 1 -10 -1007  imp:n,p=2
C This cell breaks MontePy 0.3.3
8     0 -1010  1001  1002.01003.01004.01005.01006.01007  imp:n,p=1
C Graveyard
99    0 +1010     imp:n,p=0

C surfaces
1001 SO      0.1
1002 SX -1.1 0.1
1003 SX +1.1 0.1
1004 SY -1.1 0.1
1005 SY +1.1 0.1
1006 SZ -1.1 0.1
1007 SZ +1.1 0.1
1010 SO 3

C data
C materials
C UO2 5 atpt enriched
m1        92235.80c           5
          92238.80c          95
C execution
ksrc 0 0 0
kcode 100000 1.000 50 1050
mode n p
```

**Version**

 - Version 0.3.3

**Additional context**

MontePy currently expands interpolation shortcuts and does not recompress them (#397).

### LINKED ISSUE #497: Test writing all valid MCNP input files out
With #493, writing to file is a lot less resource intensive. We should test all input files for similar issues to #172

### LINKED ISSUE #499: Shortcuts recompressed incorrectly
**Describe the bug**

Shortcuts for data entry are understood by MontePy, but they can erroneously "absorb" the first entry.

They also seem to be introducing blank lines where none belong (or maybe that is due to comments).

**To Reproduce**

``` python
import montepy
deck = montepy.read_input("testi.i")
c = deck.cells[4]
print(f"Imp:n={c.importance.neutron}, Imp:p={c.importance.photon}, Vol={c.volume}")
deck.write_problem("testo.i", overwrite=True)
deck = montepy.read_input("testo.i")
c = deck.cells[4]
print(f"Imp:n={c.importance.neutron}, Imp:p={c.importance.photon}, Vol={c.volume}")
```

**MCNP input file snippet: testi.i**

```
MCNP Test Model
C Cells
C Graveyard
99    0 +1010
C Seven spheres within a sphere
1 1 -10 -1001
2 1 -10 -1002
3 1 -10 -1003
4 1 -10 -1004
5 1 -10 -1005
6 1 -10 -1006
7 1 -10 -1007
8     0 -1010  1001  5i  1007

C surfaces
1001 SO      0.1
1002 SX -1.1 0.1
1003 SX +1.1 0.1
1004 SY -1.1 0.1
1005 SY +1.1 0.1
1006 SZ -1.1 0.1
1007 SZ +1.1 0.1
1010 SO 3

C data
C This is what the importance should expand to.
imp:n 0 1 2 3  10 12 14 16 1
C But, linear interpolation recompressed incorrectly
imp:p 0 2i 3   10 2i 16    1
C Repeat is also recompressed incorrectly
VOL   J 1 2R 2 10 2R 20    3
C materials
C UO2 5 atpt enriched
m1        92235.80c           5
          92238.80c          95
C execution
ksrc 0 0 0
kcode 100000 1.000 50 1050
mode n p

```

**Output**

The cell modifier parameters are set correctly. Here is Cell 4. However, they are written out incorrectly.

``` python
Imp:n=10.0, Imp:p=10.0, Vol=2.0

Traceback (most recent call last):
  File "./badinterp/montepy/mcnp_object.py", line 50, in __init__
    self._tree = parser.parse(input.tokenize(), input)
  File "./badinterp/montepy/input_parser/parser_base.py", line 156, in parse
    tree = super().parse(token_generator)
  File "/opt/Mamba/forge3/envs/cafe/lib/python3.9/site-packages/sly/yacc.py", line 2137, in parse
    value = p.func(self, pslice)
  File "./badinterp/montepy/input_parser/parser_base.py", line 243, in shortcut_sequence
    short_cut = syntax_node.ShortcutNode(p)
  File "./badinterp/montepy/input_parser/syntax_node.py", line 1694, in __init__
    self._expand_repeat(p)
  File "./badinterp/montepy/input_parser/syntax_node.py", line 1758, in _expand_repeat
    raise ValueError(f"Repeat cannot follow a jump. Given: {list(p)}")
ValueError: Repeat cannot follow a jump. Given: [(shortcut:Shortcuts.JUMP: [(Value, None, padding: None)]), '3R']
```

**MCNP input file snippet: testo.i**

```
MCNP Test Model
C Cells
C Graveyard
99    0 +1010
C Seven spheres within a sphere
1 1 -10 -1001 
2 1 -10 -1002 
3 1 -10 -1003 
4 1 -10 -1004 
5 1 -10 -1005 
6 1 -10 -1006 
7 1 -10 -1007 
8     0 -1010  1001  1002  1003  1004  1005  1006  1007 

C surfaces
1001 SO      0.1
1002 SX -1.1 0.1
1003 SX +1.1 0.1
1004 SY -1.1 0.1
1005 SY +1.1 0.1
1006 SZ -1.1 0.1
1007 SZ +1.1 0.1
1010 SO 3

C data
C This is what the importance should expand to.
imp:n 0 1 2 3  10 12 14 16 1
C But, linear interpolation recompressed incorrectly
     
imp:p 0 2i 3   3i 16    1
C Repeat is also recompressed incorrectly
     
C Repeat is also recompressed incorrectly
VOL   J 3R 2 10 2R 20    
C materials
C UO2 5 atpt enriched
m1        92235.80c           5
          92238.80c          95
C execution
ksrc 0 0 0
kcode 100000 1.000 50 1050
mode n p



```

**Version**

 - Versions 0.2.10 - 0.3.3 ([0ade822](https://github.com/idaholab/MontePy/commit/0ade8227cc288cb53481dd435a5a211edadc72f7))

**Additional context**

See also #489.

### LINKED ISSUE #519: Add support for reading from non-file steams
**Is your feature request related to a problem? Please describe.**

It would be nice (mainly in testing) to do:

``` python
fh = io.StringIO()
fh.write(...)
fh.seek(0)
problem = montepy.read_input(fh)
```

**Describe the solution you'd like**
`montepy.read_input` should accept file streams.

**Describe alternatives you've considered**
None that I can think of.

**Additional context**
This is mostly for testing.

### LINKED ISSUE #526: Extra blank lines are sometimes added on write_problem
**Describe the bug**

This came from #520 testing, and specifically `tests/inputs/..`: `test_universe.imcnp`, `test_interp_edge.imcnp`, `testReadEdgeCase.imcnp`.

**To Reproduce**

A short code snippet of what you have ran. Please change or remove any specific values or anything that can't be public. For example:

``` python
montepy.read_input("foo.imcnp").write_problem('bar.imcnp')
```

**MCNP input file snippet**

If applicable, please include a small section of the input file you were working on. If it includes any specific values please change or remove them. For example:

```
MCNP Test Model for MOAA
C cells
c
1 1 20
         -1000  $ dollar comment
     imp:n,p=1 trcl=5
     u=1
2 2 8
      -1005
     imp:n=1
     imp:p=0.5
     lat 1 
     fill= 0:1 0:1 0:0 1 0 0 1 (5)
3 3 -1
      1000 1005 -1010
     imp:n,p=1
99 0
      1010
     imp:n,p=0
     fill=1
5 0 
      #99
      imp:n,p=3 fill=1 (1 0.0 0.0)
c foo end comment

C surfaces
1000 SO 1
1005 RCC 0 1.5 -0.5 0 0 1 0.25
1010 SO 3

C data
C materials
C UO2 5 atpt enriched
m1        92235.80c           5 &
92238.80c          95
c testing comments
sc1 This is a high quality source comment
Fc5 Such a good tally comment.
C Iron
m2        26054.80c        5.85
          26056.80c       91.75
          26057.80c        2.12
          26058.80c        0.28
C water
m3        1001.80c           2
           8016.80c           1
MT3 lwtr.23t
TR5 0.0 0.0 1.0
C execution
ksrc 0 0 0
kcode 100000 1.000 50 1050
phys:p j 1 2j 1
mode n p
vol NO 2J 1 1.5 J
```

This writes out as:

``` 
MCNP Test Model for MOAA
C cells
c
1 1 20
         -1000  $ dollar comment
     IMP:n=0.0
2 2 8
      -1005
     imp:n=1
     imp:p=0.5 $ graveyard
3 3 -1
      1000 1005 -1010
c
c foo end comment
c
     IMP:n=0.0
99 0 1000:-1005 1010 $graveyard
     IMP:n=0.0
5 0
      #99 IMP:n=0.0

C surfaces
1000 SO 1
1005 RCC 0 1.5 -0.5 0 0 1 0.25
1010 SO 3

C data
C materials
C UO2 5 atpt enriched
u 350 2J -1
FILL 2J 350
m1        92235.80c           5 &
92238.80c          95
C Iron

m2        26054.80c        5.85
          26056.80c       91.75
          26057.80c        2.12
          26058.80c        0.28
C water
m3        1001.80c           2
           8016.80c           1
MT3 lwtr.23t
C execution
ksrc 0 0 0
kcode 100000 1.000 50 1050
phys:p j 1 2j 1
mode n p
vol NO 2J 1 1.5 J
```

**Version**

 - Version 0.4.0

**Additional context**

### PR DESCRIPTION
**Pull request by MicahGale**
# Description

## Bug fix #499
This fixes the bug from #499. This bug came down to a bug when two repeat shortcuts are touching. The second repeat assumes that the repeat node to its left is the value this one is repeating. To solve this the logic is updated to see if these edge values match.

## Test Improvement #497 
In the process I implemented testing where all valid input files are readin, "written" out and then re-read in, and the original and new files are compared. This found various padding bugs. 

## Bug fix: #489
For the bug: #489, interpolations (and all shortcuts) in geometry definitions were not recompressed. This was due to the fact that the `GeometryTree` had no idea that a shortcut was involved. This will be fixed by:

1. Making it possible to have an interpolation of integer values only.
2. Labeling `GeometryTree` accordingly when it comes from a shortcut
3. Creating a method to recompress a shortcut in a `GeoemtryTree`. 

## Bug Fix #526 

This is actually #352 in disguise. 

## Bug fix #352

This was due to the trailing comments not being found properly due to a few reasons. In #352 the issue was due to the default trees for cell modifiers being added. This makes it look like there were no trailing comments.

This case was fixed by flagging default tree items in the parameters, and ignoring them for transferring comments.

The other case found was not being transferred from a tally comment. In this case the padding was attached to the classifier. The `get_trailing_comment` logic had to be updated to allow falling through to that very first node in the tree. This issue was in part causing #526 due to the new line additions due to the comment being in the wrong object.

- [x] Fixes #499 
- [x] Fixes #519 
- [x] Fixes #497 
- [x] fixes #489
- [x] fixes #526 
- [x] Fixes #352.

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] Update unit testing to be more thurough
- [x] Cover logic case of `Jump` `Repeat`
- [x] Implement more unit tests for geometry shortcuts.

### CODE REVIEW

**Comment by MicahGale** [comment]
Not actually replicating #499.

**Comment by tjlaboss** [comment]
The first expansion is ok. It seems to occur on the $N^\text{th}$ expansion on a line where $N>1$.

**Comment by MicahGale** [comment]
Awww. I see. Thanks.

**Comment by MicahGale** [comment]
This looks more correctly failing:

``` python
>               assert new_line == gold_line.rstrip().expandtabs(8)
E               AssertionError: assert 'imp:e   0 2r 2r' == 'imp:e   0 2r 1 r'
E
E                 - imp:e   0 2r 1 r
E                 ?              ^^
E                 + imp:e   0 2r 2r
E                 ?              ^

```

File: montepy/input_parser/syntax_node.py [hunk 1]
```diff
@@ -1967,7 +1967,24 @@ def _format_jump(self):
...
+            return isinstance(node, ValueNode) or (
+                isinstance(node, type(self))
+                and (
+                    leading_node._type != Shortcuts.REPEAT
+                    or (
+                        leading_node._type == Shortcuts.REPEAT
+                        and math.isclose(self.nodes[0].value, node.nodes[-1].value)
+                    )
+                )
+            )
```
Comment (tjlaboss) [inline montepy/input_parser/syntax_node.py]: This is not very pythonic. Don't be afraid to use more `return` statements.

Comment (tjlaboss) [reply montepy/input_parser/syntax_node.py]: Or make `can_use_last_node` a Boolean variable, not a function.

File: tests/inputs/test_importance.imcnp [hunk 2]
```diff
@@ -41,7 +41,7 @@ MT3 lwtr.23t
 C execution
 ksrc 0 0 0
 imp:n,p 1 1 1 0 3
-imp:e   0 0 0 1 2
+imp:e   0 2r 1 r
```
Comment (tjlaboss) [inline tests/inputs/test_importance.imcnp:44]: We also need to test a Jump. 

The following still is not recompressed correctly:

```
C Repeat recompressed incorrectly
VOL   J 1 2R 2 10 2R 20    3   
```

Leading to

```
Traceback (most recent call last):
  File "montepy/mcnp_object.py", line 50, in __init__
    self._tree = parser.parse(input.tokenize(), input)
  File "montepy/input_parser/parser_base.py", line 156, in parse
    tree = super().parse(token_generator)
  File "/opt/Mamba/forge3/envs/cafe/lib/python3.9/site-packages/sly/yacc.py", line 2137, in parse
    value = p.func(self, pslice)
  File "montepy/input_parser/parser_base.py", line 243, in shortcut_sequence
    short_cut = syntax_node.ShortcutNode(p)
  File "montepy/input_parser/syntax_node.py", line 1694, in __init__
    self._expand_repeat(p)
  File "montepy/input_parser/syntax_node.py", line 1758, in _expand_repeat
    raise ValueError(f"Repeat cannot follow a jump. Given: {list(p)}")
ValueError: Repeat cannot follow a jump. Given: [(shortcut:Shortcuts.JUMP: [(Value, None, padding: None)]), '3R']

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "mwe.py", line 6, in <module>
    deck = montepy.read_input("testo.i")
  File "montepy/input_parser/input_reader.py", line 31, in read_input
    problem.parse_input(replace=replace)
  File "montepy/mcnp_problem.py", line 293, in parse_input
    raise e
  File "montepy/mcnp_problem.py", line 278, in parse_input
    obj = obj_parser(input)
  File "montepy/data_inputs/data_parser.py", line 46, in parse_data
    return data_class(input)
  File "montepy/data_inputs/volume.py", line 33, in __init__
    super().__init__(input, in_cell_block, key, value)
  File "montepy/data_inputs/cell_modifier.py", line 32, in __init__
    super().__init__(input, fast_parse)
  File "montepy/data_inputs/data_input.py", line 58, in __init__
    super().__init__(input, self._parser)
  File "montepy/mcnp_object.py", line 53, in __init__
    raise MalformedInputError(
montepy.errors.MalformedInputError: 
: testo.i, line 33
Error parsing object of type: <class 'montepy.data_inputs.volume.Volume'>: Repeat cannot follow a jump. Given: [(shortcut:Shortcuts.JUMP: [(Value, None, padding: None)]), '3R']

the full input:
C Repeat recompressed incorrectly
VOL   J 3R 2 10 2R 20
```

File: tests/test_integration.py [hunk 3]
```diff
@@ -481,7 +484,7 @@ def test_importance_format_unmutated(importance_problem):
     print(output)
     assert len(output) == 2
     assert "imp:n,p 1 1 1 0 3" == output[0]
-    assert "imp:e   0 0 0 1 2" == output[1]
+    assert "imp:e   0 2r 1 r" == output[1]
```
Comment (tjlaboss) [inline tests/test_integration.py:487]: see above

**Review by tjlaboss — CHANGES_REQUESTED** [review]

File: tests/inputs/test_importance.imcnp (hunk 2 above)
Comment (MicahGale) [reply tests/inputs/test_importance.imcnp:44]: See: https://github.com/idaholab/MontePy/pull/520/files#diff-86f3bbc374cf733ce672ca79b57725426ef2a38df960fc9fe2e93406792ed300R920

**Comment by MicahGale** [comment]
I accidentally stumbled upon #489 in this testing:

``` python
E               AssertionError: assert '10214   0    (1  2.03.04)' == '10214   0    (1  2i  4)'
E
E                 - 10214   0    (1  2i  4)
E                 ?                   ^^^
E                 + 10214   0    (1  2.03.04)
E                 ?                   ^^^^^
```

(this is for `tests/inputs/test_interp_edge.imcnp`).

**Comment by MicahGale** [comment]
The `ShortcutNode` -> `GeometryTree` -> `ShortcutNode` transform is a bit lossy. The question is: is this acceptable for now?

```
E               AssertionError: assert '10214   0    (1  2I 4 )' == '10214   0    (1  2i  4)'
E
E                 - 10214   0    (1  2i  4)
E                 ?                   ^^
E                 + 10214   0    (1  2I 4 )
E                 ?                   ^  +
 ```

**Comment by tjlaboss** [comment]
Yes.

1. Make it accurate
2. Make it valid
3. Make it pretty

**Comment by MicahGale** [comment]
Also found: #526, #524

File: montepy/input_parser/cell_parser.py [hunk 4]
```diff
@@ -112,19 +112,27 @@ def geometry_term(self, p):
...
         if isinstance(p.geometry_term, syntax_node.GeometryTree):
             left = p.geometry_term
+            left.mark_last_leaf_shortcut(shortcut.type)
         else:
             left = next(node_iter)
+        left_type = (
+            shortcut.type
+            if isinstance(left, syntax_node.ValueNode) and left == shortcut.nodes[0]
+            else None
+        )
```
Comment (tjlaboss) [inline montepy/input_parser/cell_parser.py:126]: consider:

```python
left_type = None
if isinstance(left, syntax_node.ValueNode) and left == shortcut.nodes[0]:
    left_type = shortcut.type
```

File: montepy/input_parser/syntax_node.py [hunk 5]
```diff
@@ -1728,6 +1832,13 @@ def end_padding(self, padding):
             )
         self._end_pad = padding
 
+    @property
+    def type(self):
+        """
+        TODO
+        """
+        return self._type
```
Comment (tjlaboss) [inline montepy/input_parser/syntax_node.py:1840]: What remains to be done?

File: montepy/input_parser/syntax_node.py [hunk 6]
```diff
@@ -1834,7 +1954,7 @@ def _can_consume_node(self, node, direction, last_edge_shortcut=False):
         :type node: ValueNode
         :param direction: the direct to go in. Must be in {-1, 1}
         :type direction: int
-        :param last_edge_shortcut: Whether or the previous node in the list was
+        :param last_edge_shortcut: Whether or not the previous node in the list was
```
Comment (tjlaboss) [inline montepy/input_parser/syntax_node.py:1957]: "Whether ~~or not~~"

File: montepy/surfaces/cylinder_on_axis.py [hunk 7]
```diff
@@ -44,7 +44,7 @@ def radius(self):
 
     def validate(self):
         super().validate()
-        if not self.radius:
+        if self.radius is None:
```
Comment (tjlaboss) [inline montepy/surfaces/cylinder_on_axis.py:47]: This change will allow a radius of 0.0, which is a valid quadric surface.

**Review by tjlaboss — CHANGES_REQUESTED** [review]
Need to test multiple consecutive shortcuts of all types. For example, the following linear interpolation is expanded incorrectly.

```
< imp:p 0 2i 3   10 2i 16    1
---      
> imp:p 0 2i 3   3i 16    1
```

#526 also persists. I'll post an example.

**Comment by tjlaboss** [comment]
#### original 

```
MCNP Test Model
C Cells
C Graveyard
99    0 +1010
C Seven spheres within a sphere
1 1 -10 -1001
2 1 -10 -1002
3 1 -10 -1003
4 1 -10 -1004
5 1 -10 -1005
6 1 -10 -1006
7 1 -10 -1007
8     0 -1010  1001  1002  1003  1004  1005  1006  1007

C surfaces
1001 SO      0.1
1002 SX -1.1 0.1
1003 SX +1.1 0.1
1004 SY -1.1 0.1
1005 SY +1.1 0.1
1006 SZ -1.1 0.1
1007 SZ +1.1 0.1
1010 SO 3

C data
C Let the importances be equal.
imp:n 0 1 2 3  10 12 14 16 1
C Second linear interpolation recompressed incorrectly
imp:p 0 2i 3   10 2i 16    1
C Repeat 
VOL   J 1 2R 2 10 2R 20    
C materials
C UO2 5 atpt enriched
m1        92235.80c           5
          92238.80c          95
C execution
ksrc 0 0 0
kcode 100000 1.000 50 1050
mode n p
```

#### result
```
MCNP Test Model
C Cells
C Graveyard
99    0 +1010
C Seven spheres within a sphere
1 1 -10 -1001
2 1 -10 -1002
3 1 -10 -1003
4 1 -10 -1004
5 1 -10 -1005
6 1 -10 -1006
7 1 -10 -1007
8     0 -1010  1001  1002  1003  1004  1005  1006  1007

C surfaces
1001 SO      0.1
1002 SX -1.1 0.1
1003 SX +1.1 0.1
1004 SY -1.1 0.1
1005 SY +1.1 0.1
1006 SZ -1.1 0.1
1007 SZ +1.1 0.1
1010 SO 3

C data
C Let the importances be equal.
imp:n 0 1 2 3  10 12 14 16 1
C Second linear interpolation recompressed incorrectly

imp:p 0 2i 3   3i 16    1
C Repeat

C Repeat
VOL   J 1 2R 2 10 2R 20
C materials
C UO2 5 atpt enriched
m1        92235.80c           5
          92238.80c          95
C execution
ksrc 0 0 0
kcode 100000 1.000 50 1050
mode n p
```

File: montepy/input_parser/syntax_node.py (hunk 5 above)
Comment (MicahGale) [reply montepy/input_parser/syntax_node.py:1840]: I still need to finish all the doc strings.

File: montepy/input_parser/syntax_node.py [hunk 8]
```diff
@@ -295,26 +304,117 @@ class GeometryTree(SyntaxNodeBase):
...
+            def start_shortcut():
+                short = ShortcutNode(short_type=short_type)
+                # give an interpolate it's old beginning to give it right
+                # start value
+                if short_type in {
+                    Shortcuts.LOG_INTERPOLATE,
+                    Shortcuts.INTERPOLATE,
+                } and isinstance(end, ShortcutNode):
+                    short.append(end.nodes[-1])
+                    short._has_pseudo_start = True
```
Comment (MicahGale) [inline montepy/input_parser/syntax_node.py:377]: Can't decide if this flag is hacky or not.

It is necessary to update `self._begin` with `load_nodes`.

**Comment by MicahGale** [comment]
Can decide if coverage needs to be increased: https://coveralls.io/builds/69589075

> 141 of 144 new or added lines in 6 files covered. (97.92%)

>1 existing line in 1 file now uncovered.
6146 of 6265 relevant lines covered (98.1%)

> 0.98 hits per line

File: montepy/input_parser/syntax_node.py (hunk 8 above)
Comment (tjlaboss) [reply montepy/input_parser/syntax_node.py:398]: a little bit but it's ok

**Comment by tjlaboss** [comment]
#526 still persists. I suggest removing that from the scope of this PR and merging the existing fixes.

**Review by tjlaboss — CHANGES_REQUESTED** [review]
https://github.com/idaholab/MontePy/issues/526 still persists. I suggest removing that from the scope of this PR and merging the existing fixes.

**Comment by MicahGale** [comment]
#526 is being tested for. Did you try it with something else?

**Comment by MicahGale** [comment]
> Need to test multiple consecutive shortcuts of all types. For example, the following linear interpolation is expanded incorrectly.
> 
> ```
> < imp:p 0 2i 3   10 2i 16    1
> ---      
> > imp:p 0 2i 3   3i 16    1
> ```
> 
> #526 also persists. I'll post an example.

I think I covered this one right?

**Comment by tjlaboss** [comment]
Try reading and rewriting this input: https://github.com/idaholab/MontePy/pull/520#issuecomment-2325199121

I'm using commit 843db181452f29c36780ec4094273c1df1d0732d

**Comment by MicahGale** [comment]
Ohh I see it now, I focused on the wrong problem there.

**Comment by MicahGale** [comment]
Ok your bug with regards to `Importance` seems to be solved.

This had to be manually addressed because `importance` is a special snow-flake and has its own system for handling syntax trees.

**Review by tjlaboss — APPROVED** [review]
